scipy/stats/_stats_py.py

```diff
@@ -4657,11 +4648,18 @@ def statistic(x, y, axis):
     r = xp.asarray(xp.clip(r, -one, one))
     r[const_xy] = xp.nan
 
-    # As explained in the docstring, the distribution of `r` under the null
-    # hypothesis is the beta distribution on (-1, 1) with a = b = n/2 - 1.
-    ab = xp.asarray(n/2 - 1)
-    dist = _SimpleBeta(ab, ab, loc=-1, scale=2)
-    pvalue = _get_pvalue(r, dist, alternative, xp=xp)
+    # Make sure we return exact 1.0 or -1.0 values for n == 2 case as promissed
+    # in the docs.
+    if n == 2:
+        r = xp_sign(r)
```

Comment:
Would this work? Probably the intent is more clear, and it would be better to use a native function of the library rather than our homegrown function `xp_sign`.
```suggestion
        r = xp.round(r)
```